dot/rpc/modules/state_test.go

```diff
@@ -280,15 +282,28 @@ func TestStateModuleGetKeysPaged(t *testing.T) {
 	}
 }
 
-// Implement Tests once function is implemented
+// TestCall tests the state_call.
+// TODO: Improve runtime tests
+// https://github.com/ChainSafe/gossamer/issues/3234
 func TestCall(t *testing.T) {
 	ctrl := gomock.NewController(t)
+	testHash := common.NewHash([]byte{0x01, 0x02})
+	rt := wasmer.NewTestInstance(t, runtime.WESTEND_RUNTIME_v0929)
 
 	mockNetworkAPI := mocks.NewMockNetworkAPI(ctrl)
 	mockStorageAPI := mocks.NewMockStorageAPI(ctrl)
-	sm := NewStateModule(mockNetworkAPI, mockStorageAPI, nil, nil)
+	mockBlockAPI := mocks.NewMockBlockAPI(ctrl)
+	mockBlockAPI.EXPECT().BestBlockHash().Return(testHash)
+	mockBlockAPI.EXPECT().GetRuntime(testHash).Return(rt, nil)
+
+	sm := NewStateModule(mockNetworkAPI, mockStorageAPI, nil, mockBlockAPI)
 
-	err := sm.Call(nil, nil, nil)
+	req := &StateCallRequest{
+		Method: "Core_version",
+		Params: "0x",
+	}
+	var res StateCallResponse
+	err := sm.Call(nil, req, &res)
```

Comment:
can we assert something in the response?  At least not zero length?

Reply:
Good idea! I've added the check